### ISSUE TITLE
MN eating up lots of RAM

### ISSUE THREAD

**Issue by G12c4** [peterhajas/MobileNotifier#30]
- mobilenotifier seems to be using abot 150 mb of RAM
- push notifications seem to be late in whatsapp

**Comment by eaferrari24** [comment]
This ram thing is killing me. I'm on a 3GS 4.2.1 in an hour or so my ram will drop from 100mb to 15 and apps start having to auto close due to a lack of ram... Would any logs help you in solving this problem? I'm having to respring hourly. And I know it's coming from MN and not anythig else on my phone

**Comment by sNick86** [comment]
Same experience here!

**Comment by timnovinger** [comment]
Peter, I think this issue can be marked as resolved. I haven't noticed this happening lately.

**Comment by Gbouck** [comment]
This does not need to be marked as resolved. Was having terrible issues with RAM before the last update, and since updating at 2:00AM on 3/16/11, I've had LESS of a problem with RAM usage than before, but still having issues, and also, since the update, my device will occasionally respring itself. For no reason. And then come back up in safe mode requiring me to then Respring again. Not sure what's going on, just wanna make sure that this is still being worked on and not marked as resolved. :-D

**Comment by eaferrari24** [comment]
I agree with Gbouck... much less of a problem when it comes to ram usage. I have seen my ram get low with the new update, but defiantly not as low as the previous update which would get to the point of closing the only app i had open (messenger).

I have had the same safe mode crash as Gbouck. I figured it was just me since I heard nobody else mention it :P so far this crash has happened to me twice.

hope you find what is causing it!
(iphone 3gs 4.2.1)

**Comment by G12c4** [comment]
I still have low ram problems, its going down slower with the new update tho :p

**Comment by peterhajas** [comment]
Not yet a resolved issue, there are still memory issues I'm working with. I need to figure out a better way - other than hooking Instruments up to SpringBoard and hoping for the best - to find memory leaks. I will not mark this as resolved until I see absolutely zero leaks. I think MNAlertManager is still leaking, but I'm quite confident that MNAlertViewController (the previous culprit) is pretty darn ironed out.

Thanks for the comments everyone, and thanks for submitting code Tim!

**Comment by Gbouck** [comment]
Thanks for everything, Peter. If there's anything I can run on my phone that'll produce any relevant memory usage information to you, I'm happy to do it. Just let me know! I'm always down for bein a guinea pig. :-)
